src/gt4py/next/iterator/ir_utils/domain_utils.py

```diff
@@ -119,18 +120,24 @@ def translate(
                     trace_shifts.Sentinel.ALL_NEIGHBORS,
                     trace_shifts.Sentinel.VALUE,
                 ]
-                # note: ugly but cheap re-computation, but should disappear
-                horizontal_sizes = _max_domain_sizes_by_location_type(offset_provider)
+                horizontal_sizes: dict[str, itir.Expr]
+                if symbolic_domain_sizes is not None:
```

Comment:
why?

Reply:
Please add a comment to explain what's going on.

Reply:
Added a docstring to infer_expr and a back reference to that docstring here.